lambdas/issuecredential/src/main/java/uk/gov/di/ipv/cri/kbv/api/service/EvidenceFactory.java

```diff
@@ -84,99 +78,67 @@ public Object[] create(KBVItem kbvItem, EvidenceRequest evidenceRequest)
         return new Map[] {objectMapper.convertValue(evidence, Map.class)};
     }
 
-    private CheckDetail[] createCheckDetailsWithKbvQuality(KBVItem kbvItem)
-            throws JsonProcessingException {
-        var questionState = objectMapper.readValue(kbvItem.getQuestionState(), QuestionState.class);
+    private CheckDetail[] withKbvQuality(KBVItem kbvItem) throws JsonProcessingException {
+        QuestionState questionState =
+                objectMapper.readValue(kbvItem.getQuestionState(), QuestionState.class);
+        KbvQuestionAnswerSummary answerSummary = kbvItem.getQuestionAnswerResultSummary();
 
-        if (hasPassedWithOneIncorrectAnswer(kbvItem)) {
+        if (hasPassedWithOneIncorrectAnswerAfterTwoBatches(kbvItem, questionState)) {
             /**
              * Scenario: We create the `checkDetails` with the lowest 3 kbvQuality values, so we are
              * excluding the highest kbvQuality value.
              *
-             * <p>1 question was wrong in the first batch, therefore 2 questions were given in
-             * second batch, and they were answered correctly. We can sort by quality and remove any
-             * additional question(s)
+             * <p>(3 out of 4 prioritised) One question was wrong in the first batch, therefore 2
+             * questions were given in second (2 out of 3 prioritised) One question was wrong in the
+             * first batch, therefore 1 question was given in second batch, and they were answered
+             * correctly. We can sort by quality and remove any additional the one the highest KBV
+             * Quality value
              */
-            if (questionState.isQuestionReceivedBatchCountEqualTo(2)) {
-                return createCheckDetailsBySortingOnKbvQuality(kbvItem, questionState);
-            }
+            return CheckDetailsBuilder.init(questionState, answerSummary, this.kbvQualityMapping)
+                    .getQuestionIdsFromAllBatches()
+                    .createCheckDetailsWithKbvQuality()
+                    .sortByKbvQualityFromLowestToHighest()
+                    .filterByNumberOfCorrectQuestions()
+                    .loadCheckDetailsToArray();
+        } else if (hasPassedWithOneIncorrectAnswerAfterThreeBatches(kbvItem, questionState)) {
             /**
              * Scenario: 2 questions in first batch were correct, third question in the second batch
              * was incorrect, and fourth question in the third batch was correct. We can safely
              * exclude the third question from the `checkDetails`
              */
-            return createCheckDetailsBySkipping3rdIncorrectQa(questionState);
+            return CheckDetailsBuilder.init(questionState, answerSummary, this.kbvQualityMapping)
+                    .getQuestionIdsButWithThe1stRemovedFromBatchTwo()
```

Comment:
Could this be split into multiple methods do you think?

Reply:
Yes. I think so